src/Season_Counter.elm

```diff
@@ -0,0 +1,97 @@
+module Season_Counter exposing (..)
+
+import Browser
+import Html exposing (div, h2, text)
+import Html.Attributes as Attributes exposing (style)
+import Task exposing (Task)
+import Time exposing (toHour, toMinute, toSecond, utc)
+
+
+type alias Model =
+    { current : Time.Posix
+    , target : Time.Posix
+    , zone : Time.Zone
+    }
+
+
+init : flags -> ( Model, Cmd Msg )
+init _ =
+    ( { current = Time.millisToPosix 0
+      , zone = Time.utc
+      , target = Time.millisToPosix {- 1573211800000 -} 1578150000000
+      }
+    , Task.perform ChangeTimeZone Time.here
+    )
+
+
+type Msg
+    = ChangeTime Time.Posix
+    | ChangeTimeZone Time.Zone
+
+
+view : Model -> Html.Html Msg
+view model =
+    let
+        currentMillis : Int
+        currentMillis =
+            Time.posixToMillis model.current
+
+        dateTimeMillis : Int
+        dateTimeMillis =
+            Time.posixToMillis model.target
+
+        millisToKickOff : Int
+        millisToKickOff =
+            dateTimeMillis - currentMillis
+
+        daysToKickOff : Int
+        daysToKickOff =
+            millisToKickOff // (60 ^ 2 * 24 * 1000)
+
+        hoursToKickOff : Int
+        hoursToKickOff =
+            modBy (60 ^ 2 * 24 * 1000) millisToKickOff // (1000 * 60 ^ 2)
+
+        minutesToKickOff : Int
+        minutesToKickOff =
+            modBy (1000 * 60 ^ 2) millisToKickOff // (1000 * 60)
+
+        secondsToKickOff : Int
+        secondsToKickOff =
+            modBy (1000 * 60) millisToKickOff // 1000
```

Comment:
If you calculate these values gradually, it'll be much simpler. You just need the modulo and the division result of each step.
Pseudocode:
```
milliseconds -> mod 1000 -> sub second milliseconds
            \
             -> div 1000 -> seconds -> mod 60 -> sub minute seconds
                                   \
                                    -> div 60 -> minutes
```
and so on.

Reply:
You can also create a helper function which gives you both the division and the modulo at the same time 😏